store/tikv/gcworker/gc_worker.go

```diff
@@ -1659,28 +1694,34 @@ func (w *GCWorker) saveValueToSysTable(key, value string) error {
 }
 
 // RunGCJob sends GC command to KV. It is exported for kv api, do not use it with GCWorker at the same time.
-func RunGCJob(ctx context.Context, s tikv.Storage, safePoint uint64, identifier string, concurrency int) error {
+func RunGCJob(ctx context.Context, s tikv.Storage, pd pd.Client, safePoint uint64, identifier string, concurrency int) error {
 	gcWorker := &GCWorker{
-		store: s,
-		uuid:  identifier,
+		store:    s,
+		uuid:     identifier,
+		pdClient: pd,
+	}
+
+	if concurrency <= 0 {
+		return errors.Errorf("[gc worker] gc concurrency should greater than 0, current concurrency: %v", concurrency)
 	}
 
 	err := gcWorker.resolveLocks(ctx, safePoint, concurrency, false)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	if concurrency <= 0 {
-		return errors.Errorf("[gc worker] gc concurrency should greater than 0, current concurrency: %v", concurrency)
-	}
-
 	err = gcWorker.saveSafePoint(gcWorker.store.GetSafePointKV(), safePoint)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	// Sleep to wait for all other tidb instances update their safepoint cache.
 	time.Sleep(gcSafePointCacheInterval)
 
+	safePoint, err = gcWorker.setTiDBServiceSafePoint(ctx, safePoint)
```

Comment:
Should this be called before `safeSafePoint`?

Get the minimum safe point first and then resolve locks and save it?